### PR TITLE
fixing wrong comment for claim_staker()

### PR DESCRIPTION
**Pull request by Maar-io**
Misleading comment for claim_staker in the Solidity interface file

**Check list**
- [ ] added unit tests
- [ ] updated documentation

### CODE REVIEW

File: precompiles/dapps-staking/DappsStaking.sol [hunk 1]
```diff
@@ -54,12 +54,16 @@ interface DappsStaking {
...
+    ///         In order to claim multiple eras, this call has to be called multiple times.
     ///         Staker account is derived from the caller address.
-    function claim_staker(address) external;
+    /// @param smart_contract smart contract address used for staking
+    function claim_staker(address smart_contract) external;
 
     /// @notice Claim one era of unclaimed dapp rewards for the specified contract and era.
-    function claim_dapp(address, uint128) external;
+    /// @param smart_contract smart contract address
+    /// @param era, The era to be claimed
```
Comment (Dinonard) [inline precompiles/dapps-staking/DappsStaking.sol]: `era, The era`?

**Comment by Dinonard** [comment]
This is good - we should consider making an example contract with usage and add it to docs and our repo.
Since this is for devs, they will probably find it easier to understand when they see concrete usage.

File: precompiles/dapps-staking/DappsStaking.sol [hunk 2]
```diff
@@ -54,12 +54,16 @@ interface DappsStaking {
     /// @notice Withdraw all funds that have completed the unbonding process.
     function withdraw_unbonded() external;
 
-    /// @notice Claim one era of unclaimed staker rewards for the specifeid contract.
+    /// @notice Claim earned staker rewards for the oldest unclaimed era.
```
Comment (Dinonard) [inline precompiles/dapps-staking/DappsStaking.sol:60]: Why not add a disclaimer on top that devs should read pallet rustdocs to better understand these functionalities?

**Review by Dinonard — CHANGES_REQUESTED** [review]

File: precompiles/dapps-staking/DappsStaking.sol (hunk 1 above)
Comment (Maar-io) [reply precompiles/dapps-staking/DappsStaking.sol]: it looks stupid here, but with syntax highlighter is ok. Let me rephrase it
Screenshot 2022-07-15 at 13 56 25

File: precompiles/dapps-staking/DappsStaking.sol (hunk 2 above)
Comment (Maar-io) [reply precompiles/dapps-staking/DappsStaking.sol:60]: added  on top of the file

File: precompiles/dapps-staking/DappsStaking.sol (hunk 1 above)
Comment (Maar-io) [reply precompiles/dapps-staking/DappsStaking.sol]: @Dinonard check now

File: precompiles/dapps-staking/DappsStaking.sol [hunk 3]
```diff
@@ -4,39 +4,42 @@ pragma solidity >=0.7.0;
 
 /// Interface to the precompiled contract on Shibuya/Shiden/Astar
 /// Predeployed at the address 0x0000000000000000000000000000000000005001
+/// For better understanding check the source code:
+/// repo: https://github.com/AstarNetwork/astar-frame
+/// code: precompiles/dapps-staking/src
```
Comment (Dinonard) [inline precompiles/dapps-staking/DappsStaking.sol]: Actually, the comment was to reference the pallet code - so please change that before the merge.
We have detailed rustdoc there.

**Review by Dinonard — APPROVED** [review]

File: precompiles/dapps-staking/DappsStaking.sol (hunk 3 above)
Comment (Maar-io) [reply precompiles/dapps-staking/DappsStaking.sol]: changed
